src/main/charts/bamboo/templates/statefulset.yaml

```diff
@@ -159,13 +159,33 @@ spec:
               protocol: TCP
             {{- include "common.jmx.port" . | nindent 12 }}
             {{- include "bamboo.additionalPorts" . | nindent 12 }}
+          {{- if .Values.bamboo.readinessProbe.enabled }}
           readinessProbe:
+            {{- if .Values.bamboo.readinessProbe.customProbe}}
+            {{- with .Values.bamboo.readinessProbe.customProbe }}
+            {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- else }}
             httpGet:
               port: {{ .Values.bamboo.ports.http }}
               path: {{ .Values.bamboo.service.contextPath }}/rest/api/latest/status
             initialDelaySeconds: {{ .Values.bamboo.readinessProbe.initialDelaySeconds }}
             periodSeconds: {{ .Values.bamboo.readinessProbe.periodSeconds }}
             failureThreshold: {{ .Values.bamboo.readinessProbe.failureThreshold }}
+            {{- end }}
+          {{- end }}
+          {{- if .Values.bamboo.livenessProbe.enabled }}
```

Comment:
Do we want to do `startupProbe` only when `livenessProbe` is enabled? Correct me if I'm wrong, it feels to be useful even when defining `startupProbe` alone without `livenessProbe`, given that `starupProbe` blocks `readinessProbe` from executing as well.

Reply:
thanks @yzha645! That's a good point. Yes, startupProbe is executed first, before the other two.  I think I've copied what was done for Jira. I'll make startup probes unconditional.

Reply:
Fixed. startupProbe is always there now (which is tcpSocker check)